internal/hud/tabview.go

```diff
@@ -1,26 +1,93 @@
 package hud
 
 import (
+	"fmt"
+
+	"github.com/gdamore/tcell"
+
 	"github.com/windmilleng/tilt/internal/hud/view"
 	"github.com/windmilleng/tilt/internal/rty"
 )
 
 type TabView struct {
 	view      view.View
 	viewState view.ViewState
+	tabState  view.TabState
 }
 
 func NewTabView(v view.View, vState view.ViewState) *TabView {
 	return &TabView{
 		view:      v,
 		viewState: vState,
+		tabState:  vState.TabState,
 	}
 }
 
 func (v *TabView) Build() rty.Component {
-	l := rty.NewConcatLayout(rty.DirHor)
+	l := rty.NewConcatLayout(rty.DirVert)
+	l.Add(v.buildTabs())
+
 	log := rty.NewTextScrollLayout("log")
-	log.Add(rty.TextString(v.view.Log.Tail(mainLogLineCount).String()))
+	log.Add(rty.TextString(v.log()))
 	l.Add(log)
+
 	return l
 }
+
+func (v *TabView) log() string {
+	var ret string
+	switch v.tabState {
+	case view.TabAllLog:
+		ret = v.view.Log.String()
+
+	case view.TabBuildLog:
+		_, resource := selectedResource(v.view, v.viewState)
+		if !resource.CurrentBuild.Empty() {
+			ret = resource.CurrentBuild.Log.String()
+		} else {
+			ret = resource.LastBuild().Log.String()
+		}
+
+	case view.TabPodLog:
+		_, resource := selectedResource(v.view, v.viewState)
+		if resource.ResourceInfo != nil {
```

Comment:
should this have a more reasonable error message when there's no resource info? e.g., the Tiltfile pod log doesn't make sense

Reply:
Just went with "no logs received", which seems good enough for now, rather than trying to get clever about which kinds of logs are expected for which resources